src/main/java/lotto/domain/Lotto.java

```diff
@@ -0,0 +1,65 @@
+package lotto.domain;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class Lotto {
+    public static final int LOTTO_NUMBER_UPPER_BOUND = 45;
+
+    private List<Integer> numbers;
+
+    public Lotto() {
+        List<Integer> lottoNumberCandidate = new ArrayList<>();
+        for (int i = 1; i <= LOTTO_NUMBER_UPPER_BOUND; ++i) {
+            lottoNumberCandidate.add(i);
+        }
+        Collections.shuffle(lottoNumberCandidate);
+        this.numbers = sortNumbers(cropNumbers(lottoNumberCandidate));
+    }
+
+    public Lotto(List<Integer> numbers) {
+        this.numbers = sortNumbers(numbers);
+    }
+
+    private List<Integer> cropNumbers(List<Integer> numbers) {
+        return numbers.subList(0, 6);
+    }
+
+    private List<Integer> sortNumbers(List<Integer> numbers) {
+        Collections.sort(numbers);
+        return numbers;
+    }
+
+    public Division compareNumbers(Lotto lotto) {
+        int matchCount = 0;
+        for (int number: lotto.numbers) {
+            matchCount += isInNumbers(number);
+        }
```

Comment:
```suggestion
        long matchCount = numbers.stream()
                .filter(number -> lotto.numbers.contains(number))
                .count();
```
와 같이 stream을 적절히 활용해볼 수 있을 것 같아요 :)